### PR TITLE
declared-license-mapping: Relax LGPL mapping if no version is specified

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Review by oheger-bosch — COMMENTED** [review]
Changes look good to me, but as I am no domain expert, I would appreciate opinions from others.

**Comment by sschuberth** [comment]
FYI, the analyzer test differences seem to stem from some strange and unrelated Maven Central vs. JCenter issues.

File: spdx-utils/src/main/resources/declared-license-mapping.yml [hunk 1]
```diff
@@ -244,7 +244,7 @@
 "GNU LGPL v3+": LGPL-3.0-or-later
 "GNU LGPL": LGPL-2.1-only
 "GNU Lesser General Public Licence": LGPL-2.1-only
-"GNU Lesser General Public License (LGPL)": LGPL-2.1-only
+"GNU Lesser General Public License (LGPL)": LGPL-2.1-or-later
```
Comment (mnonnenmacher) [inline spdx-utils/src/main/resources/declared-license-mapping.yml:247]: Commit message type: "**if** no version"

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

**Comment by mnonnenmacher** [comment]
> FYI, the analyzer test differences seem to stem from some strange and unrelated Maven Central vs. JCenter issues.

Are you sure? I didn't check, but I wouldn't be surprised if the changed mapping is the reason for the failing tests.

**Comment by sschuberth** [comment]
> Are you sure? I didn't check, but I wouldn't be surprised if the changed mapping is the reason for the failing tests.

Yes, these tests also failed locally for me on a different branch:

image

**Review by tsteenbe — CHANGES_REQUESTED** [review]
> "If the Library does not specify a license version
number, you may choose any version ever published by the Free Software
Foundation", so at least any later version should be considered. As a
result, change mappings for declared licenses without a version from
"LGPL-2.1-only" to "LGPL-2.1-or-later"

 I would include a note in the commit message that we choice or-later mapping for LGPL starting from version 2.1 as 2.0 is rarely used.

**Comment by sschuberth** [comment]
> I would include a note in the commit message that we choice or-later mapping for LGPL starting from version 2.1 as 2.0 is rarely used.

Hmm, that's more a rationale for why it was 2.1 already before, than about the "-only" vs "-or-later" change in this commit. But I'll add it anyway for reference / as a reminder.

File: spdx-utils/src/main/resources/declared-license-mapping.yml [hunk 2]
```diff
@@ -255,7 +255,7 @@
 "GNU Lesser General Public License v3 (LGPLv3)": LGPL-3.0-only
 "GNU Lesser General Public License v3 or later (LGPLv3+)": LGPL-3.0-or-later
 "GNU Lesser General Public License v3+": LGPL-3.0-or-later
-"GNU Lesser General Public License": LGPL-2.1-only
+"GNU Lesser General Public License": LGPL-2.1-or-later
 "GNU Lesser General Public License, Version 2.1": LGPL-2.1-only
 "GNU Lesser Public License": LGPL-2.1-only
 "GNU Library or Lesser General Public License (LGPL)": LGPL-2.1-only
```
Comment (tsteenbe) [inline spdx-utils/src/main/resources/declared-license-mapping.yml:261]: Why did you not change these to  LGPL-2.1-or-later?

**Review by tsteenbe — CHANGES_REQUESTED** [review]

File: spdx-utils/src/main/resources/declared-license-mapping.yml (hunk 2 above)
Comment (sschuberth) [reply spdx-utils/src/main/resources/declared-license-mapping.yml]: Simply an oversight. I now when through all mapping to `LGPL-2.1-only` and found a bunch of more to fix.

**Review by tsteenbe — DISMISSED** [review]

**Comment by sschuberth** [comment]
In the latest update I just had to adapt some tests.

**Review by tsteenbe — APPROVED** [review]

**Pull request merged**